### PR TITLE
Include puppet scripts for APIM 2.0.0 fully distributed setup. (5 profiles, 5 nodes)

### PR DESCRIPTION
**Pull request by chamindias**
Note : 

With the newly introduced "traffic manager" profile in API Manager 2.0.0, there is a requirement to support file deletion in puppet scripts. Once this new profile is introduced, it is needed to remove existing webapps and jaggeryapps from the <APIM_HOME>/repository/deployment/server directory. Otherwise it will throw exceptions at the startup of the traffic manager node.
If we remove the required files manually (by logging in to traffic manager node, execute "rm -r webapps" and "rm -r jaggeryapps" and restarting it), those exceptions are not going to be there.
Please refer [1] for more information.

[1] https://docs.wso2.com/display/CLUSTER44x/Clustering+API+Manager+2.0.0#ClusteringAPIManager2.0.0-ConfiguringtheTrafficManager

JIRA for this : https://wso2.org/jira/browse/PMODULES-24

### CODE REVIEW

File: hieradata/dev/wso2/wso2am/2.0.0/default/api-key-manager.yaml [hunk 1]
```diff
@@ -25,6 +25,10 @@ wso2::apim_traffic_manager:
   username: "%{hiera('wso2::super_admin::username')}"
   password: "%{hiera('wso2::super_admin::password')}"
 
+wso2::template_list:
+  - repository/conf/axis2/axis2_TM.xml
+  - repository/conf/registry_TM.xml
+
```
Comment (pubudu538) [inline hieradata/dev/wso2/wso2am/2.0.0/default/api-key-manager.yaml]: We don't need to define these two templates. These templates will be referred in embedded files.

File: hieradata/dev/wso2/wso2am/2.0.0/default/traffic-manager.yaml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+# limitations under the License.
+
+---
+wso2::mgt_hostname: tm.dev.wso2.org
+wso2::hostname: tm.dev.wso2.org
+
+wso2::template_list:
+  - repository/conf/axis2/axis2.xml
+  - repository/conf/registry.xml
+
```
Comment (pubudu538) [inline hieradata/dev/wso2/wso2am/2.0.0/default/traffic-manager.yaml]: Please remove these two files.

File: hieradata/dev/wso2/wso2am/2.0.0/default/api-key-manager.yaml (hunk 1 above)
Comment (chamindias) [inline hieradata/dev/wso2/wso2am/2.0.0/default/api-key-manager.yaml]: Fixed : https://github.com/wso2/puppet-modules/pull/87/commits/04dd34927096ab148e508db85f62de407f812108

File: hieradata/dev/wso2/wso2am/2.0.0/default/traffic-manager.yaml (hunk 2 above)
Comment (chamindias) [inline hieradata/dev/wso2/wso2am/2.0.0/default/traffic-manager.yaml]: Fixed : https://github.com/wso2/puppet-modules/pull/87/commits/05d72504960932fba81d1bd860604eab361b7e14

**Pull request merged**